org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/HoverHandlerTest.java

```diff
@@ -133,8 +141,26 @@ public void testHoverVariable() throws Exception {
 
 		//then
 		assertNotNull(hover);
-		String result = hover.getContents().get(0).getLeft();//wow this is so elegant!
-		assertEquals("Unexpected hover "+result, "String[] args - java.Foo.main(String[])", result);
+		assertNotNull(hover.getContents());
+		MarkedString signature = hover.getContents().get(0).getRight();
+		assertEquals("Unexpected hover " + signature, "java", signature.getLanguage());
+		assertEquals("Unexpected hover " + signature, "String[] args - java.Foo.main(String[])", signature.getValue());
+	}
+
+	@Test
+	public void testHoverMethod() throws Exception {
+		// given
+		// Hovers on the overriding print()
+		String payload = createHoverRequest("src/java/Foo.java", 8, 15);
+		TextDocumentPositionParams position = getParams(payload);
+
+		// when
+		Hover hover = handler.hover(position, monitor);
+
+		// then
+		assertNotNull(hover);
+		MarkedString result = hover.getContents().get(0).getRight();
+		assertEquals("Unexpected hover ", "void print(String arg0)", result.getValue());
```

Comment:
This test fails if the JDK running the tests has sources available (Oracle JDK typically). OpenJDK doesn't install sources by default.
I suggest you pattern-match against `String \\w+` instead.